src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.Mono.cs

```diff
@@ -47,1846 +46,2030 @@
-			get {
-				return 0;
-			}
-		}
-
-		public override MethodBase DeclaringMethod {
-			get {
-				return null;
-			}
-		}
-
-		public GenericTypeParameterBuilder[] DefineGenericParameters (params string[] names)
-		{
-			if (names == null)
-				throw new ArgumentNullException ("names");
-			if (names.Length == 0)
-				throw new ArgumentException ("names");
-
-			generic_params = new GenericTypeParameterBuilder [names.Length];
-			for (int i = 0; i < names.Length; i++) {
-				string item = names [i];
-				if (item == null)
-					throw new ArgumentNullException ("names");
-				generic_params [i] = new GenericTypeParameterBuilder (this, null, item, i);
-			}
-
-			return generic_params;
-		}
-
-		public static ConstructorInfo GetConstructor (Type type, ConstructorInfo constructor)
-		{
-			/*FIXME I would expect the same checks of GetMethod here*/
-			if (type == null)
-				throw new ArgumentException ("Type is not generic", "type");
-
-			if (!type.IsGenericType)
-				throw new ArgumentException ("Type is not a generic type", "type");
-
-			if (type.IsGenericTypeDefinition)
-				throw new ArgumentException ("Type cannot be a generic type definition", "type");
-
-			if (constructor == null)
-				throw new NullReferenceException (); //MS raises this instead of an ArgumentNullException
-
-			if (!constructor.DeclaringType.IsGenericTypeDefinition)
-				throw new ArgumentException ("constructor declaring type is not a generic type definition", "constructor");
-			if (constructor.DeclaringType != type.GetGenericTypeDefinition ())
-				throw new ArgumentException ("constructor declaring type is not the generic type definition of type", "constructor");
-
-			ConstructorInfo res = type.GetConstructor (constructor);
-			if (res == null)
-				throw new ArgumentException ("constructor not found");
-
-			return res;
-		}
-
-		static bool IsValidGetMethodType (Type type)
-		{
-			if (type is TypeBuilder || type is TypeBuilderInstantiation)
-				return true;
-			/*GetMethod() must work with TypeBuilders after CreateType() was called.*/
-			if (type.Module is ModuleBuilder)
-				return true;
-			if (type.IsGenericParameter)
-				return false;
-
-			Type[] inst = type.GetGenericArguments ();
-			if (inst == null)
-				return false;
-			for (int i = 0; i < inst.Length; ++i) {
-				if (IsValidGetMethodType (inst [i]))
-					return true;
-			}
-			return false;
-		}
-
-		public static MethodInfo GetMethod (Type type, MethodInfo method)
-		{
-			if (!IsValidGetMethodType (type))
-				throw new ArgumentException ("type is not TypeBuilder but " + type.GetType (), "type");
-
-			if (type is TypeBuilder && type.ContainsGenericParameters)
-				type = type.MakeGenericType (type.GetGenericArguments ());
-
-			if (!type.IsGenericType)
-				throw new ArgumentException ("type is not a generic type", "type");
-
-			if (!method.DeclaringType.IsGenericTypeDefinition)
-				throw new ArgumentException ("method declaring type is not a generic type definition", "method");
```

Comment:
do we have an issue already to move strings into resources? or in most cases, find the existing resources

Reply:
Not sure.  It'd be good to do, though.  There are probably several passes that should be made through the mono-specific corelib sources.  This PR is just enough to turn on our analyzers.

Reply:
I'll open one

Reply:
https://github.com/dotnet/runtime/issues/34056